### PR TITLE
add env variable to change CRAN url

### PR DESCRIPTION
**Pull request by maelle**
cf #18

### CODE REVIEW

File: R/urls.R [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
 r_svn_url <- "https://svn.r-project.org/R/tags/"
-r_download_url <- "https://cran.r-project.org/src/base/R-3/"
-r_win_download_url <- "https://cran.r-project.org/bin/windows/base/"
-r_macos_download_url <- "https://cran.r-project.org/bin/macosx/"
+
+if (nzchar(Sys.getenv("R_RVERSIONS_CRAN_URL"))){
+  cran_url <- Sys.getenv("R_RVERSIONS_CRAN_URL")
+} else {
+  cran_url <- "cran.r-project.org"
+}
```
Comment (gaborcsardi) [inline R/urls.R:15]: This is running at install time, which is not what we want I guess. We'll need to put these into a function.

File: R/urls.R [hunk 2]
```diff
@@ -1,12 +1,23 @@
+r_svn_url <- function(){
+  "https://svn.r-project.org/R/tags/"
```
Comment (gaborcsardi) [inline R/urls.R:2]: can you pls leave a space after `()`? thanks.

File: R/urls.R [hunk 3]
```diff
@@ -1,12 +1,23 @@
...
+}
 
-r_svn_url <- "https://svn.r-project.org/R/tags/"
+cran_url <- function(){
```
Comment (gaborcsardi) [inline R/urls.R]: space

File: R/urls.R [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
+  if (nzchar(Sys.getenv("R_RVERSIONS_CRAN_URL"))){
```
Comment (gaborcsardi) [inline R/urls.R:6]: space here as well, before `{`

File: R/urls.R [hunk 5]
```diff
@@ -1,12 +1,23 @@
...
+  } else {
+    "cran.r-project.org"
+  }
+}
 
-if (nzchar(Sys.getenv("R_RVERSIONS_CRAN_URL"))){
-  cran_url <- Sys.getenv("R_RVERSIONS_CRAN_URL")
-} else {
-  cran_url <- "cran.r-project.org"
+r_download_url <- function(){
```
Comment (gaborcsardi) [inline R/urls.R]: space

File: R/urls.R [hunk 6]
```diff
@@ -1,12 +1,23 @@
...
+  paste0(cran_url(), "/src/base/R-3/")
 }
 
-r_download_url <- paste0(cran_url, "/src/base/R-3/")
-r_win_download_url <- paste0(cran_url, "/bin/windows/base/")
-r_macos_download_url <- paste0(cran_url, "/bin/macosx/")
+r_win_download_url <- function(){
```
Comment (gaborcsardi) [inline R/urls.R]: space

File: R/urls.R [hunk 7]
```diff
@@ -1,12 +1,23 @@
...
+  paste0(cran_url(), "/bin/windows/base/")
+}
+
+r_macos_download_url <- function(){
```
Comment (gaborcsardi) [inline R/urls.R]: space

**Review by gaborcsardi — COMMENTED** [review]
Thanks! Some formatting nitpick and then it is good. :)

**Comment by maelle** [comment]
Ok! Before closing the issue can you please add the environment variable to the Travis and Appveyor projects?

**Comment by gaborcsardi** [comment]
Right, can you actually add it to the `.travis.yml` and `appveyor.yml` files?

File: .travis.yml [hunk 8]
```diff
@@ -2,6 +2,9 @@ language: r
 sudo: false
 cache: packages
 
+env:
+  - R_RVERSIONS_CRAN_URL='cloud.r-project.org'
```
Comment (gaborcsardi) [inline .travis.yml]: Don't we need the `https://`?

Comment (maelle) [reply .travis.yml]: apparently not, it worked :joy: I added it.

**Comment by gaborcsardi** [comment]
I think the default is http, so it is better to add it..... I'll wait for the CI and then merge.

**Pull request merged**